root/etc/e-smith/templates/etc/httpd/conf.d/default-virtualhost.inc/15user_settings

```diff
@@ -0,0 +1,55 @@
+#
+# 15 user_settings
+#
+{
+    # check if user settings is enabled and redirect url is set
+    if (
+        (${'cockpit.socket'}{'UserSettingsPage'} || 'disabled') eq 'enabled' &&
+        (${'cockpit.socket'}{'UserSettingsPageAlias'} || '') ne ''
+    ) {
+        $ProxyConf = qq(
+ProxyPreserveHost On
+ProxyRequests Off
+
+# enable proxypass for ws and http
+ProxyPass /cockpit/socket ws://127.0.0.1:9090/cockpit/socket
+ProxyPassReverse /cockpit/socket ws://127.0.0.1:9090/socket
+ProxyPass /cockpit http://127.0.0.1:9090/cockpit
+ProxyPassReverse /cockpit http://127.0.0.1:9090
+
+);
+        $RedirectConf = qq(
+    # create redirect path to user settings
+    Redirect ${'cockpit.socket'}{'UserSettingsPageAlias'} /cockpit/\@localhost/nethserver/index.html#/settings
+);
+        $OUT .= $ProxyConf;
+
+        # restrict access if necessary
+        $OUT .= "# enable location to restric access if necessary\n";
+        $OUT .= "<Location />\n";
+
+        # check if external access is enabled
+        if ((${'cockpit.socket'}{'UserSettingsGrantAccess'} || 'disabled') eq 'enabled') {
+
+            # check if external ips are set
+            if ((${'cockpit.socket'}{'UserSettingsGrantIPs'} || '') ne '') {
+                # split ips
+                my @ips = split(',', ${'cockpit.socket'}{'UserSettingsGrantIPs'});
+
+                # create require ip directive
+                foreach my $ip (@ips) {
+                    $OUT .= "    Require ip $ip\n";
```

Comment:
Does it make sense to (conditionally) add green and trusted networks too?

Reply:
For the `Server Manager` access part on port `9090` we don't include Trusted Networks. I think we can use the same behaviour here

Reply:
We should see this page as a separate web application, I agree with @DavidePrincipi to support the trusted networks.
My proposal is:
- remove UserSettingsGrantIPs prop
- use local networks (see [roundcube](https://github.com/NethServer/nethserver-roundcubemail/tree/master/root/etc/e-smith/templates/etc/httpd/conf.d/roundcubemail.conf) as example)

Reply:
Ok so if we enable external access, the only ips granted are the ips in the trusted networks list.